components/director/internal/selfregmanager/selfreg_manager.go

```diff
@@ -81,7 +94,7 @@ func (s *selfRegisterManager) IsSelfRegistrationFlow(ctx context.Context, labels
 func (s *selfRegisterManager) PrepareForSelfRegistration(ctx context.Context, resourceType resource.Type, labels map[string]interface{}, id string, validate func() error) (map[string]interface{}, error) {
```

Comment:
Now `PrepareForSelfRegistration` is executed inside of the DB transaction because of labels but I believe we wanted to be outside and not to have an external call part of it? Is this considered during implementation or if there is another approach to keep the external call outside of the DB transact? 

Maybe extracting `s.createSelfRegPrepRequest` from the `PrepareForSelfRegistration ` and executing it before the DB transact and afterward getting/checking region label and if it doesn't exist to defer/rollback "self reg preparation"?

Reply:
Discussed and agreed to move the logic for deriving the tenant region into the tenant mapping hydrator.
Dependency: https://github.com/kyma-incubator/compass/pull/2521 to get merged

Reply:
@PetarTodorovv - I've adopted this branch/PR as we spoke:
1. Merged your branch into this one
2. Ensured that all obj ctx providers assign a region
3. In the `addConsumersToExtra` functionality we double check that all matched obj ctx providers have the same region assigned
4. Also in the `addConsumersToExtra` functionality make sure to assign the discovered region as `reqData.Body.Extra`

I'll wait for your PR to officially reach `main` branch and then this PR can be moved forward.